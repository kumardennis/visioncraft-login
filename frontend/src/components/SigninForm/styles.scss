h2 {
  color: white;
}
.signin-form {
  background: linear-gradient(
    to bottom right,
    rgba(0, 0, 0, 0.8),
    rgba(25, 56, 48, 0.8)
  );
  padding: 2rem;
  width: 400px;
  height: fit-content;
  box-shadow: 0 0 20px 10px black;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;

  margin: auto;

  @media (max-width: 1024px) {
    width: 90%;
    margin: 2rem 0;
  }

  form {
    display: flex;
    flex-direction: column;

    label {
      margin: 3rem 0 1rem 0;
      z-index: 10;
      color: white;
      font-size: larger;
      border: none;
      width: fit-content;
    }

    .user-input {
      padding: 1.5rem;
      border: 2px solid lightgrey;
      border-radius: 1rem;
      background-color: transparent;
      color: white;

      &::placeholder {
        color: rgb(107, 107, 107);
      }

      &:focus {
        outline: none;
      }
    }

    #submit-sign-in {
      background: linear-gradient(to bottom right, #28e4c4, #0b867c);
      box-shadow: 0 0 5px darkblue;
      border: none;
      outline: none;
      padding: 1rem;
      margin: 2rem 0;
      color: white;
      font-weight: bold;
      border-radius: 50px;
    }
  }

  .toc {
    margin: 2rem;
    color: lightgrey;
    text-align: center;
  }
}
