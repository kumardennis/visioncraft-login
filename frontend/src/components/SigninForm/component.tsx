import React, { useState, useEffect } from "react";
import { MdEmail } from "react-icons/md";
import { AiTwotoneLock } from "react-icons/ai";
import { BsArrowBarLeft } from "react-icons/bs";
import "./styles.scss";

type SigninFormProps = {
  handleIsSigningUp: () => void;
  handleScreen: () => void;
};

interface SigninData {
  email: string;
  password: string;
}

interface Response {
  type: string;
  message: string;
}

const SigninForm = ({ handleIsSigningUp, handleScreen }: SigninFormProps) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [signinData, setSigninData] = useState<SigninData | {}>({});

  const handleInput = (event: React.FormEvent<HTMLInputElement>) => {
    switch (event.currentTarget.id) {
      case "user-email":
        setEmail(event.currentTarget.value);
        break;

      case "user-password":
        setPassword(event.currentTarget.value);
        break;

      default:
        break;
    }
  };

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();

    setSigninData({
      email,
      password,
    });
  };

  useEffect(() => {
    if (Object.keys(signinData).length !== 0) {
      fetch("/signin", {
        method: "POST",
        headers: {
          "Content-Type": "application/json",
        },
        body: JSON.stringify(signinData),
      })
        .then((response) => response.json())
        .then((data: Response) =>
          alert(data.message + " Click 'Okay' to continue!")
        )
        .then(handleScreen)
        .catch((err: Error) => alert(err));
    }
  }, [signinData]);

  return (
    <div className="signin-form">
      <h2>Sign in!</h2>
      <form onSubmit={handleSubmit}>
        <label htmlFor="user-email">
          Email <MdEmail />
        </label>
        <input
          className="user-input"
          id="user-email"
          name="user-email"
          type="email"
          placeholder="Eg: abc123@gmail.com"
          onChange={handleInput}
          value={email}
          required
        />

        <label htmlFor="user-password">
          Password <AiTwotoneLock />
        </label>
        <input
          className="user-input"
          id="user-password"
          name="user-password"
          type="password"
          placeholder="Eg: somecrazyAs123Password"
          onChange={handleInput}
          value={password}
          required
        />

        <input id="submit-sign-in" type="submit" value="Sign in!" />
      </form>

      <div className="toc">
        Use of this service is subject to terms and conditions.
      </div>

      <button onClick={handleIsSigningUp} className="change-auth-screen">
        <BsArrowBarLeft /> Take me to Sign Up!
      </button>
    </div>
  );
};

export default SigninForm;
