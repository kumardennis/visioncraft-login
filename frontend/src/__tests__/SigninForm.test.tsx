import React from "react";
import { mount, shallow, configure } from "enzyme";
import SigninForm from "../components/SigninForm/component";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Signin Form", () => {
  it("should render SignupForm", () => {
    const signinForm = shallow(
      <SigninForm changeAuthForm={() => {}} handleScreen={() => {}} />
    );
    expect(signinForm).toMatchSnapshot();
  });
});

describe("Inputs for SigninForm", () => {
  let wrapper;

  test("Render email check", () => {
    wrapper = mount(
      <SigninForm changeAuthForm={() => {}} handleScreen={() => {}} />
    );

    const emailInput = wrapper
      .find('input[id="user-email"]')
      .simulate("change");

    expect(emailInput).toMatchSnapshot();
  });

  test("Render password check", () => {
    wrapper = mount(
      <SigninForm changeAuthForm={() => {}} handleScreen={() => {}} />
    );

    const passwordInput = wrapper
      .find('input[id="user-password"]')
      .simulate("change");

    expect(passwordInput).toMatchSnapshot();
  });
});
