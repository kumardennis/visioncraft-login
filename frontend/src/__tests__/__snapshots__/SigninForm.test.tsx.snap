// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs for SigninForm Render email check 1`] = `ReactWrapper {}`;

exports[`Inputs for SigninForm Render password check 1`] = `ReactWrapper {}`;

exports[`Inputs for SignupForm Render email check 1`] = `ReactWrapper {}`;

exports[`Inputs for SignupForm Render password check 1`] = `ReactWrapper {}`;

exports[`Signin Form should render SignupForm 1`] = `ShallowWrapper {}`;
